### DISCUSSION TITLE
milvus goexit

### DISCUSSION THREAD

**Discussion by huangxueliang** [milvus-io/milvus#37522] [Q&A and General discussion]
goroutine 81198606 [select]:
runtime.gopark(0xc002ca66f0?, 0x3?, 0x30?, 0x0?, 0xc002ca669a?)
        /usr/local/go/src/runtime/proc.go:398 +0xce fp=0xc002ca6510 sp=0xc002ca64f0 pc=0x1de6b0e
runtime.selectgo(0xc002ca66f0, 0xc002ca6694, 0xc005f8dec0?, 0x0, 0xc002ca67a8?, 0x1)
        /usr/local/go/src/runtime/select.go:327 +0x725 fp=0xc002ca6630 sp=0xc002ca6510 pc=0x1df7565
github.com/milvus-io/milvus/pkg/mq/msgstream.(*MqTtMsgStream).consumeToTtMsg(0xc004f6eeb0, {0x5eeef20, 0xc003df0740})
        /workspace/source/pkg/mq/msgstream/mq_msgstream.go:778 +0x165 fp=0xc002ca67b8 sp=0xc002ca6630 pc=0x3e45045
github.com/milvus-io/milvus/pkg/mq/msgstream.(*MqTtMsgStream).bufMsgPackToChannel.func4()
        /workspace/source/pkg/mq/msgstream/mq_msgstream.go:676 +0x28 fp=0xc002ca67e0 sp=0xc002ca67b8 pc=0x3e44e48
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1650 +0x1 fp=0xc002ca67e8 sp=0xc002ca67e0 pc=0x1e1c341
created by github.com/milvus-io/milvus/pkg/mq/msgstream.(*MqTtMsgStream).bufMsgPackToChannel in goroutine 12758
        /workspace/source/pkg/mq/msgstream/mq_msgstream.go:676 +0xb7f

goroutine 80466226 [select]:
runtime.gopark(0xc00266be30?, 0x2?, 0x18?, 0xbd?, 0xc00266bdc4?)
        /usr/local/go/src/runtime/proc.go:398 +0xce fp=0xc00266bc50 sp=0xc00266bc30 pc=0x1de6b0e
runtime.selectgo(0xc00266be30, 0xc00266bdc0, 0xc003389fb0?, 0x0, 0x0?, 0x1)
        /usr/local/go/src/runtime/select.go:327 +0x725 fp=0xc00266bd70 sp=0xc00266bc50 pc=0x1df7565
github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*Dispatcher).work(0xc003a7d500)
        /workspace/source/pkg/mq/msgdispatcher/dispatcher.go:209 +0x2a6 fp=0xc00266bfc8 sp=0xc00266bd70 pc=0x45d1d86
github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*Dispatcher).Handle.func3()
        /workspace/source/pkg/mq/msgdispatcher/dispatcher.go:191 +0x25 fp=0xc00266bfe0 sp=0xc00266bfc8 pc=0x45d1a45
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1650 +0x1 fp=0xc00266bfe8 sp=0xc00266bfe0 pc=0x1e1c341
created by github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*Dispatcher).Handle in goroutine 80466161
        /workspace/source/pkg/mq/msgdispatcher/dispatcher.go:191 +0x4bb

rax    0x0
rbx    0x7fe22cb7f700
rcx    0xffffffffffffffff
rdx    0x0
rdi    0x2
rsi    0x7fe22cb5ee00
rbp    0x7fe2455a0918
rsp    0x7fe22cb5ee00
r8     0x0
r9     0x7fe22cb5ee00
r10    0x8
r11    0x246
r12    0x7fe22a6304e0
r13    0x7fe2bfbe70b0
r14    0x8b
r15    0x7fe229700540
rip    0x7fe2bafe503b
rflags 0x246
cs     0x33
fs     0x0
gs     0x0

-----

**Comment by yhmo** [comment]
Do you have the full log?
Which version of your milvus?
What happened?

**Reply by huangxueliang** [reply]
1.Do you have the full log? 
i have no the full log,because my milvus has re-deploy, as flow:
docker-compose down
docker-compose up -d

2.milvus:v2.4.6

3.i not do anything

**Reply by huangxueliang** [reply]
![aa](https://github.com/user-attachments/assets/63a5e7b0-eaf9-47dd-bf1c-b16a1ad1520e)

**Reply by yhmo** [reply]
Do you mean the milvus standalone crashes suddenly?

So far as I know, there are two fixes for crashes in 2.4.8 and 2.4.15, you can try upgrading to the latest version:

Fixed in 2.4.8
- Fixed crashes caused by illegal search expressions. ([#35307](https://github.com/milvus-io/milvus/pull/35307))

Fixed in 2.4.15
- Resolved a deadlock issue if the QueryNode crashed during shard client initialization ([#37354](https://github.com/milvus-io/milvus/pull/37354)).
